src/Compilers/CSharp/Portable/Symbols/Synthesized/TypeSubstitutedLocalSymbol.cs

```diff
@@ -97,5 +97,15 @@ internal override ImmutableArray<Diagnostic> GetConstantValueDiagnostics(BoundEx
         {
             return _originalVariable.GetConstantValueDiagnostics(boundInitValue);
         }
+
+        internal TypeSubstitutedLocalSymbol WithSynthesizedLocalKindAndSyntax(SynthesizedLocalKind kind, SyntaxNode syntax)
```

Comment:
I considered that, but thought it would be a bit of an overkill. It is certainly possible.